.github/workflows/terragrunt-apply-production.yml

```
@@ -49,6 +49,19 @@ jobs:
         with:
           is-tagged: 'true'
 
+  generate-lambda-functions-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
```

Could you have this job checkout the version of the code being deployed.  This is just to cover an edge case where the version we want to deploy has a different set of lambda function code:
```suggestion
    needs: get-version
    runs-on: ubuntu-latest
    env:
      VERSION: ${{ needs.get-version.outputs.version }}
    steps:
      - name: Checkout
        uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
        with:
          ref: ${{ env.VERSION }}
```

---

Oh yes absolutely!